src/marqo/vespa/vespa_client.py

```diff
@@ -206,59 +219,58 @@ def _gzip_compress(self, directory: str) -> io.BytesIO:
         byte_stream.seek(0)
         return byte_stream
 
-    def _create_deploy_session(self) -> int:
+    def _create_deploy_session(self, httpx_client: httpx.Client) -> int:
         endpoint = f'{self.config_url}/application/v2/tenant/default/session?from=' \
                    f'{self.config_url}/application/v2/tenant/default/application/default/environment' \
                    f'/default/region/default/instance/default'
 
-        response = httpx.post(endpoint)
+        response = httpx_client.post(endpoint)
 
         self._raise_for_status(response)
 
         return response.json()['session-id']
 
-    def _download_application(self, session_id: int) -> str:
+    def _download_application(self, session_id: int, httpx_client: httpx.Client) -> str:
```

Comment:
Good idea. Though to be honest this is fragile as it is, seeing as we're counting on the single connection pool to help route all downloads to the same config server (but connection pooling is best effort and not guaranteed). So I'd rather run it like this at least for a while, before adding extra complexity (in case we encounter unexpected behaviour). Also this is strictly for index creation/modification, which isn't a common operation.

Reply:
makes sense